Feature/Check DB Table Exists

MatthewCane opened the pull request:
Resolves #444.

- Introduced `--expect-table` option to check for the existence of a database table before completing
- Works for MySQL and Postgres
- Added tests for scenarios where a specified table exists and does not exist

atkrad commented:
Hi @MatthewCane,

Thanks for your contribution! I think it would be better to rename the `--table-exists` flag to `--expect-table`, to keep the naming consistent with other commands like HTTP, Redis and etc. What do you think?

Also, it looks like the MySQL unit tests failed. Could you please check that?

atkrad commented:
@MatthewCane, regarding the unit tests — please let me know if you need any help.

MatthewCane commented:
Thank you for the feedback @atkrad, I agree on the flag naming and I have adjusted it accordingly. I have hopefully fixed the tests and added a small extra line to the setup of the database test containers to hopefully make the tests more reliable, I've had a few flakey failures:

`testcontainers.WithWaitStrategy(wait.ForListeningPort("XXXXX"))`

MatthewCane commented:
Thank you for catching those! This feature is going to be really useful for me so I'm very happy I could contribute.
